eloquentjs/8_2_locked_box.js

```diff
@@ -0,0 +1,37 @@
+const box = {
+    locked: true,
+    unlock() { this.locked = false; },
+    lock() { this.locked = true;  },
+    _content: [],
+    get content() {
+      if (this.locked) throw new Error("Locked!");
+      return this._content;
+    }
+  };
+  
+  function withBoxUnlocked(body) {
+    let initialStatus = box.locked;
+
+    try {
+        box.unlock();
+        body();
+        box.locked = initialStatus;
+    } catch(e) {
```

Comment:
For stuff like this it's worth thinking about if you want to propagate the error by throwing e within the catch block. There's no right answer here since that's contextual, I just want you to have that idea in your head.